lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/TeamViewer.pm

```diff
@@ -64,20 +74,28 @@ sub _getID {
 
         GLPI::Agent::Tools::Win32->use();
 
-        my $clientid = getRegistryValue(
-            path   => is64bit() ?
-                "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/TeamViewer/ClientID" :
-                "HKEY_LOCAL_MACHINE/Software/TeamViewer/ClientID",
+        my $clientid = getRegistryKey(                
+                path => "HKEY_LOCAL_MACHINE/SOFTWARE/TeamViewer",            
         );
+        if(!$clientid){
+            my $key = getRegistryKey(
+                path => "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/TeamViewer",
+            );
+        }  
 
         unless ($clientid) {
-            my $teamviever_reg = getRegistryKey(
-                path => is64bit() ?
-                    "HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/TeamViewer" :
-                    "HKEY_LOCAL_MACHINE/SOFTWARE/TeamViewer",
+            my $teamviever_reg = getRegistryKey(                
+                path => "HKEY_LOCAL_MACHINE/SOFTWARE/TeamViewer",  
                 # Important for remote inventory optimization
-                required    => [ qw/ClientID/ ],
+                required    => [ qw/ClientID/ ],          
             );
+            if(!$teamviever_reg){
```

Comment:
As I said earlier, you should check OS is 64 bits before looking for `Wow6432Node` node.
```suggestion
            if (!$teamviever_reg && is64bit()) {
```